src/index.js

```diff
@@ -0,0 +1,254 @@
+    return index + 1;
+  };
+
+  Evaluator.prototype.nextLeft = function (index) {
+    return index - 1;
+  };
+
+  Evaluator.prototype.nextBelow = function (index) {
+    return index + this.board.dimension;
+  };
+
+  Evaluator.prototype.nextLeftDiagonal = function (index) {
+    return this.nextLeft(this.nextBelow(index));
+  };
+
+  Evaluator.prototype.nextRightDiagonal = function (index) {
+    return this.nextRight(this.nextBelow(index));
+  };
+
+  Evaluator.prototype.getRow = function () {
+    const rowIndex = parseInt(this.lastMove / this.board.dimension, 10);
+
+    return this.lastMove % this.board.dimension === 0 ? rowIndex - 1 : rowIndex;
+  };
+
+  Evaluator.prototype.getColumn = function () {
+    const columnIndex = parseInt(this.lastMove % this.board.dimension, 10);
+    return this.lastMove % this.board.dimension === 0
+      ? this.board.dimension - 1
+      : columnIndex - 1;
+  };
+
+  Evaluator.prototype.getDiagnoal = function () {
+    const column = this.getColumn();
+    const row = this.getRow();
+
+    if (column === row) {
+      return 0;
+    }
+
+    if (column + row + 1 === this.board.dimension) {
+      return this.board.dimension - 1;
+    }
+
+    return null;
+  };
+
+  Evaluator.prototype.isOnDiagonal = function () {
+    const columnIndex = this.getColumn();
+    const rowIndex = this.getRow();
+
+    return (
+      columnIndex === rowIndex
+      || columnIndex + rowIndex + 1 === this.board.dimension
+    );
+  };
+
+  return Evaluator;
+}());
+
+const Board = (function () {
+  function Board(dimension = 3) {
+    this.dimension = dimension;
+    this.board = new Array(dimension * dimension).fill('');
+  }
+
+  Board.prototype.get = function (index) {
+    return this.board[index];
+  };
+
+  Board.prototype.put = function (item, index) {
+    this.board[index] = item;
+  };
+
+  Board.prototype.isFilled = function () {
+    return this.board.filter(e => e).length === (this.dimension * this.dimension);
+  };
+
+  return Board;
+}());
+
+const Game = (function () {
+  function Game(board) {
+    this.board = board;
+    this.playerX = null;
+    this.playerO = null;
+    this.currentPlayer = null;
+    this.finished = false;
+  }
+
+  Game.prototype.render = function () {
+    const cells = document.querySelectorAll('.cell');
+    cells.forEach((el, i) => {
+      el.textContent = this.board.get(i);
+    });
+  };
+
+  function hideAnnouncement() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'none';
+  }
+
+  function displayWinner() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'block';
+    announcementNode.textContent = `${this.currentPlayer.name} wins!`;
+  }
+
+  function displayDraw() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'block';
+    announcementNode.textContent = 'Draw!';
+  }
```

Comment:
Let's separate the DOM from application logic, you can create another file for DOM.

![image](https://user-images.githubusercontent.com/31897434/73420341-70e18b80-4322-11ea-840a-fa14e04b592b.png)

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

Reply:
Resolved